File: src/interfaces/dto/category.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class SubcategoryDto {
  @ApiProperty({ description: '고유 식별자', example: 11 })
  id!: number;

  @ApiProperty({ description: '서브카테고리 이름', example: '프론트엔드' })
  name!: string;

  @ApiProperty({ description: '서브카테고리 코드', example: 'frontend' })
  code!: string;

  @ApiProperty({ description: '생성 일시', example: '2025-05-06T12:00:00.000Z' })
  createdAt!: Date;

  @ApiProperty({ description: '수정 일시', example: '2025-05-06T12:30:00.000Z' })
  updatedAt!: Date;
}

export class CategoryDto {
  @ApiProperty({ description: '고유 식별자', example: 1 })
  id!: number;

  @ApiProperty({ description: '카테고리 이름', example: 'IT/기술' })
  name!: string;

  @ApiProperty({ description: '카테고리 코드', example: 'tech' })
  code!: string;

  @ApiProperty({ description: '생성 일시', example: '2025-05-06T12:00:00.000Z' })
  createdAt!: Date;

  @ApiProperty({ description: '수정 일시', example: '2025-05-06T12:30:00.000Z' })
  updatedAt!: Date;

  @ApiProperty({
    description: '하위 카테고리 리스트',
    type: [SubcategoryDto],
  })
  subcategories!: SubcategoryDto[];
}

File: common/src/main/java/com/katchup/common/category/dto/CategoryDto.java
package com.katchup.common.category.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 카테고리 DTO
 * 
 * 카테고리 정보를 클라이언트에 전달하기 위한 DTO입니다.
 * 카테고리의 기본 정보와 하위 카테고리 목록을 포함합니다.
 * 
 * @author katchup
 * @since 1.0.0
 */
@Getter
@AllArgsConstructor
public class CategoryDto {
    /** 카테고리 ID */
    private Long id;

    /** 카테고리 이름 */
    private String name;

    /** 카테고리 코드 */
    private String code;

    /** 생성 일시 */
    private LocalDateTime createdAt;

    /** 수정 일시 */
    private LocalDateTime updatedAt;

    /** 하위 카테고리 목록 */
    private List<SubcategoryDto> subcategories;
} 